TcUnit-Runner/VisualStudioInstance.cs

```diff
@@ -127,35 +131,61 @@ private void LoadDevelopmentToolsEnvironment(string visualStudioVersion, bool is
-            
-            // If the version is pinned check if the pinned version is available
-            if (isTcVersionPinned)
+
+            string latestTcVersion = "";
+            // Check if Version is installed
+            foreach (var possibleVersion in remoteManager.Versions)
             {
-                log.Info("Project has a pinned version: " + this.tcVersion);
-                foreach (var possibleVersion in remoteManager.Versions)
+                latestTcVersion = Char.ToString(possibleVersion[0]);
+                if (possibleVersion == this.tcVersion)
                 {
-                    if (possibleVersion == this.tcVersion)
-                    {
-                        isVersionAvailable = true;
-                        break;
-                    }
+                    isVersionAvailable = true;
                 }
-                if (isVersionAvailable)
-                    log.Info("The pinned TwinCAT version is available ");
-                else
+                // Check if optional Version is installed
+                if (possibleVersion == this.forceToThisTwinCatVersion && this.forceToThisTwinCatVersion != null)
                 {
-                    log.Error("The pinned TwinCAT version is not available ");
-                    throw new Exception("The pinned TwinCAT version is not available");
+                    isForceVersionAvailable = true;
+                    log.Info("The optional TC Version is available ");
                 }
             }
 
-            if (!isTcVersionPinned || isVersionAvailable)
+            // If the Version is Pinned, check if the Pinned Version is available
+            if (isTcVersionPinned && (String.IsNullOrEmpty(this.forceToThisTwinCatVersion)))
+            {
+
+                  if (isVersionAvailable)
+                  {
+                      log.Info("The pinned TC Version is available ");
+                  }
+                  else
+                  {
+                      log.Error("The pinned TC Version is not available ");
+                      throw new Exception("The pinned TC Version is not available");
+                  }
+             }
+
+
+            if (isTcVersionPinned && isVersionAvailable && (String.IsNullOrEmpty(this.forceToThisTwinCatVersion)))
             {
                 remoteManager.Version = this.tcVersion;
-                log.Info("Load TwinCAT version: "+ remoteManager.Version);
             }
+            else if (isForceVersionAvailable)
+            {
+                log.Info("TC Version is forced to " + this.forceToThisTwinCatVersion);
+                remoteManager.Version = this.forceToThisTwinCatVersion;
+            }
+            else if (!isForceVersionAvailable && (!String.IsNullOrEmpty(this.forceToThisTwinCatVersion)))
+            {
+                log.Error("The Optional TC Version "+ this.forceToThisTwinCatVersion + " is not available ");
```

Comment:
Minor comment: I think we need to be consistent with the upper and lowercase according to previous code...
so "The optional TwinCAT version .... is not available".

Reply:
changed